contracts/JBXBuybackDelegate.sol

```diff
@@ -308,31 +409,34 @@ contract JBXBuybackDelegate is IJBFundingCycleDataSource, IJBPayDelegate, IUnisw
                 _useReservedRate: true
             });
 
-            // 3) Burn the non-reserve token which are now left in this address (can be 0) -> result: 0 here, reservedToken in reserve
-            uint256 _nonReservedTokenInContract = _amountReceived - _reservedToken;
-
-            if (_nonReservedTokenInContract != 0) {
-                controller.burnTokensOf({
-                    _holder: address(this),
-                    _projectId: _data.projectId,
-                    _tokenCount: _nonReservedTokenInContract,
-                    _memo: "",
-                    _preferClaimedTokens: false
-                });
-            }
+            // Burn all the token received here (kept as reserved from the swap + minted just above)
+            // ie when _preferClaimed is true, burn starts with the claimed token, then continue with unclaimed ones
+            controller.burnTokensOf({
+                _holder: address(this),
+                _projectId: _data.projectId,
+                _tokenCount: _amountReceived,
+                _memo: "",
+                _preferClaimedTokens: true
+            });
+        }
+
+        // refund any extra eth left back to the beneficiary
```

Comment:
I would rather put it at the very end of didPay (if not, I think one can reenter to have an amountReceived == 0 on the second pass and force mint -> twice the token)

Reply:
yeah good point I was so into reproducing this test that i misplaced fixed now